torch/testing/_internal/common_utils.py

```diff
@@ -881,6 +881,40 @@ def safeCoalesce(self, t):
 
         return tg
 
+    # Compares the given Torch and NumPy functions on the given tensor-like object.
+    #   if copy_np_result is True then NumPy arrays will be copied before being
+    #     compared with the torch tensor. This is useful if the NumPy array has
+    #     negative strides, which would cause the import to PyTorch to fail.
+    # TODO: support bfloat16 comparisons
+    def compare_with_numpy(self, torch_fn, np_fn, tensor_like, device, dtype, *,
+                           copy_np_result=False):
+        assert TEST_NUMPY
+        assert dtype is not torch.bfloat16
+
+        a = None
+        t = None
+        if isinstance(tensor_like, torch.Tensor):
+            a = tensor_like.detach().cpu().numpy()
+            t = tensor_like
+        else:
+            a = np.array(tensor_like, dtype=torch_to_numpy_dtype_dict[dtype])
+            t = torch.tensor(tensor_like, device=device, dtype=dtype)
+
+        torch_result = torch_fn(t).cpu()
+        np_result = np_fn(a)
+
+        # Coerces NumPy results (where appropriate)
+        if isinstance(torch_result, torch.Tensor) and not isinstance(np_result, np.ndarray):
+            np_result = np.array(np_result)
+        if isinstance(np_result, np.ndarray):
+            np_result = torch.from_numpy(np_result)
+
+        # Copies the NumPy result (if requested)
+        if copy_np_result and isinstance(np_result, np.ndarray):
```

Comment:
try to torch.from_numpy, if failure try to copy and repeat instead of taking a kwarg for this

Reply:
(and move comment near that code)